core/app/models/spree/order/checkout.rb

```diff
@@ -1,338 +1,338 @@
-            before_transition to: :complete, do: :ensure_line_items_are_in_stock
+              before_transition to: :complete, do: :ensure_line_item_variants_are_not_deleted
+              before_transition to: :complete, do: :ensure_line_items_are_in_stock
 
-            after_transition to: :complete, do: :finalize!
-            after_transition to: :resumed,  do: :after_resume
-            after_transition to: :canceled, do: :after_cancel
+              after_transition to: :complete, do: :finalize!
+              after_transition to: :resumed, do: :after_resume
+              after_transition to: :canceled, do: :after_cancel
 
-            after_transition from: any - :cart, to: any - [:confirm, :complete] do |order|
-              order.update_totals
-              order.persist_totals
+              after_transition from: any - :cart, to: any - [:confirm, :complete] do |order|
+                order.update_totals
+                order.persist_totals
+              end
             end
-          end
-
-          alias_method :save_state, :save
-        end
 
-        def self.go_to_state(name, options = {})
-          self.checkout_steps[name] = options
-          previous_states.each do |state|
-            add_transition({ from: state, to: name }.merge(options))
+            alias_method :save_state, :save
           end
-          if options[:if]
-            previous_states << name
-          else
-            self.previous_states = [name]
-          end
-        end
 
-        def self.insert_checkout_step(name, options = {})
-          before = options.delete(:before)
-          after = options.delete(:after) unless before
-          after = self.checkout_steps.keys.last unless before || after
-
-          cloned_steps = self.checkout_steps.clone
-          cloned_removed_transitions = self.removed_transitions.clone
-          checkout_flow do
-            cloned_steps.each_pair do |key, value|
-              go_to_state(name, options) if key == before
-              go_to_state(key, value)
-              go_to_state(name, options) if key == after
+          def self.go_to_state(name, options = {})
+            self.checkout_steps[name] = options
+            previous_states.each do |state|
+              add_transition({ from: state, to: name }.merge(options))
             end
-            cloned_removed_transitions.each do |transition|
-              remove_transition(transition)
+            if options[:if]
+              previous_states << name
+            else
+              self.previous_states = [name]
             end
           end
-        end
 
-        def self.remove_checkout_step(name)
-          cloned_steps = self.checkout_steps.clone
-          cloned_removed_transitions = self.removed_transitions.clone
-          checkout_flow do
-            cloned_steps.each_pair do |key, value|
-              go_to_state(key, value) unless key == name
+          def self.insert_checkout_step(name, options = {})
```

Comment:
Assignment Branch Condition size for insert_checkout_step is too high. [19.34/15]